mrp_project/models/mrp_production.py

```diff
@@ -16,3 +16,20 @@ def _onchange_project_id(self):
         if not self.project_id:
             self.project_sequence = 0
 
+    @api.onchange("project_id")
+    def _onchange_project_id_parent(self):
+        for record in self:
+            if record.project_id:
+                def assign_project_to_children(mrp):
+                    child_mrp = mrp._get_children()
+                    for child in child_mrp:
+                        child.project_id = mrp.project_id
+                        assign_project_to_children(child)
+                assign_project_to_children(record)
+            else:
+                def remove_project_from_children(mrp):
+                    child_mrp = mrp._get_children()
+                    for child in child_mrp:
+                        child.project_id = False
+                        remove_project_from_children(child)
+                remove_project_from_children(record)
```

Comment:
This could be simplified IMO (assigning or removing project for a MO is the same operation IMO), could you check?
```suggestion
   def _assign_project_to_children():
       self.ensure_one()
       child_mrp = self._get_children()
       for child in child_mrp:
           child.project_id = mrp.project_id
           child._assign_project_to_children()

    @api.onchange("project_id")
    def _onchange_project_id_parent(self):
        self._assign_project_to_children()
```